staging/src/k8s.io/client-go/transport/websocket/roundtripper.go

```diff
@@ -108,7 +109,23 @@ func (rt *RoundTripper) RoundTrip(request *http.Request) (retResp *http.Response
 	}
 	wsConn, resp, err := dialer.DialContext(request.Context(), request.URL.String(), request.Header)
 	if err != nil {
-		return nil, &httpstream.UpgradeFailureError{Cause: err}
+		if err == gwebsocket.ErrBadHandshake {
```

Comment:
Should probably be `errors.As(err, &badHandshake)` to account for wrapped errors.